qa-tests-backend/src/test/groovy/ImageSignatureVerificationTest.groovy

```diff
@@ -206,7 +208,11 @@ w9e2Azq1OYIh/pbeBMHARDrBaqqmuMR9+BfAaPAYdkNTU6f58M2zBbuL0A==
     def "Check violations of policy '#policyName' for deployment '#deployment.name'"() {
         expect:
         "Verify deployment has expected violations"
-        assert waitForViolation(deployment.name, policyName, WAIT_FOR_VIOLATION_TIMEOUT) == expectViolations
+        if (expectViolations) {
+            assert waitForViolation(deployment.name, policyName, WAIT_FOR_VIOLATION_TIMEOUT) == expectViolations
```

Comment:
How about sleeping in setup spec once and here just query for the result without or with minimal timeout?

Reply:
I was thinking about that, but didn't want to introduce this due to potential flakes when i.e. reprocessing of one particular image takes longer (for whatever reason) than the other.
I'd propose to stay defensive right now and not introduce a potential to flake and potentially change this later on if we feel the need for it.

WDYT?